feat/#11 카카오 소셜 로그인 구현

eunseo-76 opened the pull request:
커밋 나눠서 해야 하는데 작업하다 보니 ㅠㅠ
담부턴 주의하겠습니다...

**구현 내용**
1. 카카오 인증 서버에서 카카오 로그인 페이지 url 받기
- 추후 로그인 버튼 클릭하면 이 url 로 이동시켜서 로그인 시킬 것임

2. (사용자가 카카오에 로그인 한 이후) 카카오 인증 서버에서 카카오 로그인 인증 코드를 받은 후  다시 access token, refresh token 받기
- access token 까지 받아야 카카오 로그인이 완료됨 

3. 카카오로 로그인한 사용자의 카카오 고유 id 값 받기
- 이 id 값으로 사용자를 구별할 것임

상기 3가지 postman 으로 테스트 완료 했습니다. 근데 프론트랑 연결 어케 하지 8ㅅ8

eunseo-76 committed:
feat: #11 카카오 소셜 로그인 시 액세스 토큰 반환까지 구현

구현 내용
- 카카오 로그인 url 받기
- 카카오 인증 코드 받기
- 카카오 액세스 토큰
구현해야 할 내용
- 사용자 정보 받기

eunseo-76 committed:
feat: #11 카카오 소셜 로그인 사용자 정보 가져오기 구현 완료

일단 카카오 사용자의 카카오 고유 id 값 가져오는 것까지는 구현

we4sley commented on MENDITTZO-Backend/src/main/java/com/mendittzo/auth/query/service/KakaoAuthService.java:
```diff
@@ -0,0 +1,106 @@
+    public KakaoLoginUrlResponseDTO getKakaoLoginUrl() {
+
+        // todo: 확인용. 나중에 삭제
+        System.out.println("yml: clientId: " + clientId);
```
이건 나중에 삭제하는 건가요?

eunseo-76 replied:
네 주석으로 todo: 이렇게 해 놓은 건 한 번에 다 지우려고요

we4sley commented on MENDITTZO-Backend/src/main/java/com/mendittzo/auth/query/service/KakaoAuthService.java:
```diff
@@ -0,0 +1,106 @@
+        System.out.println("DTO: clientId: " + loginUrlRequest.getClientId());
+        System.out.println("DTO: redirectUri: " + loginUrlRequest.getRedirectUri());
+
+        String loginUrl = "https://kauth.kakao.com/oauth/authorize?client_id="
```
url도 상수 처리해주세요

eunseo-76 replied:
넷!

we4sley commented on MENDITTZO-Backend/src/main/java/com/mendittzo/user/query/controller/UserQueryController.java:
```diff
@@ -0,0 +1,12 @@
+@RequiredArgsConstructor
+@RequestMapping("/users")
+public class UserQueryController {
+
```
클래스 생성만 하고 아직 코드 작성 안 한거죠?

eunseo-76 replied:
넵... 소셜 로그인 하다가 잘 안 풀려서 잠깐 다른 거 손 댔더니 섞였네요.....

we4sley reviewed (commented):
Spring Security 적용하고 나서 카카오에서 받아온 토큰을 토대로 저희 서비스 자체 AccessToken, RefreshToken을 발급해서 인증/인가를 진행하는 방식으로 개발할 예정이신가요?

eunseo-76 committed:
feat: #11 url 상수 처리

eunseo-76 committed:
merge: origin/develop 의 내용을 feat: #11 로 merge

eunseo-76 committed:
feat: #14 User 관련

eunseo-76 committed:
feat: 카카오 로그인 백-프론트 연결

일단 spring security, 서비스 자체 토큰 등등 다 빼고 카카오 인증 서버에서 url, 인증코드, accessToken 받아올 수 있는지 확인 완료

eunseo-76 committed:
bug: #14 DB에 닉네임만 insert 안 되는 문제 해결

DB에는 nickname 인데 DTO는 nickName 이었음

eunseo-76 committed:
feat: #14 비밀 키 생성용
